### ISSUE TITLE
Initial Plugin install creates '~' directory in the user's home directory

### ISSUE THREAD

**Issue by jchrisfarris** [turbot/steampipe#2528]
**Describe the bug**
For some reason, when steampipe is initialized on Ubuntu, an unused directory `~/~/.steampipe` is created. This empty directory is harmless but looks suspicious to any security investigator. 

**Steampipe version (`steampipe -v`)**
steampipe version 0.17.0-alpha.23

**To reproduce**
Install the [steampipe_linux_arm64.deb](https://github.com/turbot/steampipe/releases/download/v0.17.0-alpha.23/steampipe_linux_arm64.deb) on an Ubuntu 22.04 instance
run `steampipe plugin install aws`
Notice a new directory is created '~'

**Expected behavior**
Steampipe and the plugin are properly initialized in the `~/.steampipe` directory. the `~/~/.steampipe` directory is created but not populated. 

**Additional context**
```bash
ubuntu@ip-10-64-10-229:~$ ls
steampipe.deb
ubuntu@ip-10-64-10-229:~$ steampipe plugin install aws

aws                  [====================================================================] Done

Installed plugin: aws@latest v0.78.0
Documentation:    https://hub.steampipe.io/plugins/turbot/aws

ubuntu@ip-10-64-10-229:~$ ls
 steampipe.deb  '~'
ubuntu@ip-10-64-10-229:~$ find '~'
~
~/.steampipe
ubuntu@ip-10-64-10-229:~$ ls -al .steampipe/
total 28
drwxr-xr-x 7 ubuntu ubuntu 4096 Oct 13 21:47 .
drwxr-x--- 6 ubuntu ubuntu 4096 Oct 13 21:47 ..
drwxr-xr-x 2 ubuntu ubuntu 4096 Oct 13 21:47 config
drwxr-xr-x 3 ubuntu ubuntu 4096 Oct 13 21:47 db
drwxr-xr-x 2 ubuntu ubuntu 4096 Oct 13 21:47 internal
drwxr-xr-x 2 ubuntu ubuntu 4096 Oct 13 21:47 logs
drwxr-xr-x 3 ubuntu ubuntu 4096 Oct 13 21:47 plugins
ubuntu@ip-10-64-10-229:~$ steampipe --version
steampipe version 0.17.0-alpha.23
ubuntu@ip-10-64-10-229:~$ rm -rf ~/~/.steampipe/
ubuntu@ip-10-64-10-229:~$ rm -rf ~/~
ubuntu@ip-10-64-10-229:~$ ls
steampipe.deb
ubuntu@ip-10-64-10-229:~$
```

If I saw a fake '~' directory on one of my machines, I'd begin an investigation. That _looks_ exactly how an attacker would attempt to hide malware or a rootkit.

**Comment by jchrisfarris** [comment]
FWIW, this seems to also be the case on AmazonLinux with the rpm
```
[ec2-user@ip-10-64-10-116 ~]$ curl -o steampipe.rpm -L -s https://github.com/turbot/steampipe/releases/download/v0.17.0-alpha.23/steampipe_linux_arm64.rpm
[ec2-user@ip-10-64-10-116 ~]$ ls
steampipe.rpm
[ec2-user@ip-10-64-10-116 ~]$ ls -l
total 13032
-rw-rw-r-- 1 ec2-user ec2-user 13343820 Oct 13 22:00 steampipe.rpm
[ec2-user@ip-10-64-10-116 ~]$ sudo rpm -ivh steampipe.rpm
Preparing...                          ################################# [100%]
Updating / installing...
   1:steampipe-0:0.17.0~alpha.23-1    ################################# [100%]
[ec2-user@ip-10-64-10-116 ~]$ ls
steampipe.rpm
[ec2-user@ip-10-64-10-116 ~]$ steampipe plugin install aws

aws                  [====================================================================] Done

Installed plugin: aws@latest v0.78.0
Documentation:    https://hub.steampipe.io/plugins/turbot/aws

[ec2-user@ip-10-64-10-116 ~]$ ls
~  steampipe.rpm
[ec2-user@ip-10-64-10-116 ~]$
```

**Comment by e-gineer** [comment]
Good catch @jchrisfarris.

Quick note that I'm not seeing this behavior for Steampipe installed via shell script on a fresh ubuntu docker. Maybe it's something in the rpm / deb packaging process?

```
nathan@a2fb6c5256c5:~$ ls
nathan@a2fb6c5256c5:~$ 
nathan@a2fb6c5256c5:~$ steampipe query
Welcome to Steampipe v0.16.4
For more information, type .help
> 
> 
nathan@a2fb6c5256c5:~$ ls -al
total 24
drwxr-xr-x 3 nathan nathan 4096 Oct 13 23:36 .
drwxr-xr-x 1 root   root   4096 Oct 13 23:34 ..
-rw-r--r-- 1 nathan nathan  220 Oct 13 23:34 .bash_logout
-rw-r--r-- 1 nathan nathan 3771 Oct 13 23:34 .bashrc
-rw-r--r-- 1 nathan nathan  807 Oct 13 23:34 .profile
drwxr-xr-x 7 nathan nathan 4096 Oct 13 23:36 .steampipe
nathan@a2fb6c5256c5:~$ 
nathan@a2fb6c5256c5:~$ steampipe plugin install net

net                  [====================================================================] Done                

Installed plugin: net@latest v0.8.0
Documentation:    https://hub.steampipe.io/plugins/turbot/net

nathan@a2fb6c5256c5:~$ 
nathan@a2fb6c5256c5:~$ ls -al
total 24
drwxr-xr-x 3 nathan nathan 4096 Oct 13 23:36 .
drwxr-xr-x 1 root   root   4096 Oct 13 23:34 ..
-rw-r--r-- 1 nathan nathan  220 Oct 13 23:34 .bash_logout
-rw-r--r-- 1 nathan nathan 3771 Oct 13 23:34 .bashrc
-rw-r--r-- 1 nathan nathan  807 Oct 13 23:34 .profile
drwxr-xr-x 7 nathan nathan 4096 Oct 13 23:36 .steampipe
nathan@a2fb6c5256c5:~$ 
```

**Comment by jchrisfarris** [comment]
Weird. It's not coming from the plugin install command
```
[ec2-user@ip-10-64-10-116 ~]$ rm -rf '~' .steampipe/
[ec2-user@ip-10-64-10-116 ~]$ ls
generate_config.sh  steampipe.rpm
[ec2-user@ip-10-64-10-116 ~]$ steampipe query
Welcome to Steampipe v0.17.0-alpha.23
For more information, type .help
>
[ec2-user@ip-10-64-10-116 ~]$ ls
~  generate_config.sh  steampipe.rpm
[ec2-user@ip-10-64-10-116 ~]$
```

**Comment by jchrisfarris** [comment]
I tried the arm64 .tgz file and it was the same behavior

**Comment by e-gineer** [comment]
I tested with v0.16.4 ... perhaps this is only happening in the v0.17.0-alpha.x series?

**Comment by jchrisfarris** [comment]
I just confirmed it's on the MacOS version too.

**Comment by alebcay** [comment]
The issue is in https://github.com/turbot/steampipe/blob/6557737c90c774313ef24a565fa1c70ff3d49ccb/cmd/root.go#L285-L292 which was introduced in https://github.com/turbot/steampipe/commit/34f9b1ebc45f93a9e59f4129e08130880621b2b9 which is currently only on https://github.com/turbot/steampipe/pull/2518 (v0.17.0-alpha.23). This has not made it to either `main` or `v0.17.x` branches.

In lines 288-291 in the excerpt above, all occurrences of `installDir` should be `filepaths.SteampipeDir`, which should have expanded the `~` as home directory. I would open a PR but there's not exactly anything to open a PR against at the moment 🙂

**Comment by e-gineer** [comment]
Nice investigation @alebcay - thank you! Easy to get fixed from here :-)

We normally work off main, but I think in the push (pun intended) to a pre-weekend alpha it got missed - sorry about that.

**Comment by kaidaguerre** [comment]
Thanks @alebcay, good detective work.

I've fixed this bug directly on the feature branch 2510-add-workspaceprofile-and-parse-from-config 

In the end I rationalised the `~` expansion to ensure data was sanitised _on the way in_  to the viper keystore: https://github.com/turbot/steampipe/blob/8c719ab2d14a7685374c1983263c20dfabce0cd7/pkg/cmdconfig/viper.go#L31

**Comment by kaidaguerre** [comment]
fixed by a19e9d7c02e442c545e19e1a0d8d4d2e32056f69
